AWSPinpoint/AWSPinpointTargetingClient.m

```diff
@@ -74,9 +74,9 @@ - (instancetype) initWithContext:(AWSPinpointContext *) context {
 
 - (void)initGlobalAttributes {
     NSData *customAttributesData = [_keychain dataForKey:AWSPinpointEndpointAttributesKey];
-    if (customAttributesData == nil) {
-        _globalAttributes = [[NSMutableDictionary alloc] init];
-    } else {
+    NSMutableDictionary *attributes = [_context.configuration.userDefaults objectForKey:AWSPinpointEndpointAttributesKey];
```

Comment:
nit: I'd suggest use a more descriptive name that reflects what's going on, like `legacyStoredAttributes` or similar